src/components/DvdRichard.tsx

```diff
@@ -0,0 +1,56 @@
+import { useDvdScreensaver } from "react-dvd-screensaver";
+import dvdRichardPng from "../assets/dvd-richard.png";
+import { useEffect, useState } from "react";
+
+export interface DvdRichardProps {
+    /** size of the image that will be bouncing around */
+    size?: number;
+    /** colors to cycle through to tint the image with */
+    colors?: string[];
+}
+
+export function DvdRichard({ 
+    size = 300,
+    colors = [
+        "#e81416",
+        "#ffa500",
+        "#faeb36",
+        "#79c314",
+        "#487de7",
+        "#4b369d",
+        "#70369d",
+    ]
+}: DvdRichardProps) {
+    const { containerRef, elementRef, impactCount } = useDvdScreensaver({
+        speed: 5,
+    });
+    const [colorIndex, setColorIndex] = useState(0);
+
+    // Grabs a new color on each impact
+    useEffect(() => {
+        let newColorIndex = Math.floor(Math.random() * colors.length);
+        
+        // Getting new color if the new calculated index is already picked
+        if (colorIndex === newColorIndex) {
+            newColorIndex = newColorIndex - 1;
+            if (newColorIndex < 0) {
+                newColorIndex = colors.length - 1;
+            }
+        }
+        setColorIndex(newColorIndex);
+    }, [impactCount])
+
+    return (
+        <div
+            ref={containerRef}
+            style={{
+                width: window.innerWidth,
+                height: window.innerHeight,
+            }}
+        >
+            <div ref={elementRef} style={{ width: size, height: size, backgroundColor: colors[colorIndex] }}>
```

Comment:
Same question here

Reply:
This one here uses variables to adjust the css (specifically the size and the color) which is easier to do inline in typescript than doing it in css files

Reply:
So, if this function ends up making a dvd.css file, would you still keep that inline css there or move it to dvd.css? At what point does it become better practice to create a css file

Reply:
When looking it up, it looks like css files are slightly (maybe to the point of being negligently) more performative, mainly with the idea that react just has to add a class name to the component versus adding all the style elements to the component. 

I personally just prefer keeping everything in react, which could be inline or just defined as an object in another typescript file and we just pass it to the style prop.